backends/ebpf/ebpftargets.py

```diff
@@ -168,58 +193,67 @@ def interface_of_filename(self, f):
                    lstrip(self.pcapPrefix).rsplit('_', 1)[0])
 
     def compile_p4(self, argv):
-        ''' To override '''
-        ''' Compile the p4 target '''
+        """ To override """
+        """ Compile the p4 target """
         if not os.path.isfile(self.options.p4filename):
             raise Exception("No such file " + self.options.p4filename)
-        args = ["./p4c-ebpf"]
-        args.extend(["--target", self.options.target])
-        args.extend(["-o", self.template])
-        args.extend(argv)
-        result, errtext = run_timeout(self.options, args, TIMEOUT, self.stderr)
+        ebpfdir = os.path.dirname(__file__)
+
+        # Initialize arguments for the makefile
+        args = self.get_make_args(ebpfdir, self.options.target)
+        # name of the output source file
+        args.append("BPFOBJ=" + self.template + ".c")
+        # location of the P4 input file
+        args.append("P4FILE=" + self.options.p4filename)
+        p4_args = ' '.join(map(str, argv))
+        if (p4_args):
+            # Remaining arguments
+            args.append("P4ARGS=\"" + p4_args + "\"")
+        errmsg = "Failed to compile P4:"
+        result = run_timeout(self.options, args, TIMEOUT,
+                             self.outputs, errmsg)
         if result != SUCCESS:
-            reportError("Error %d: Failed to compile P4." %
-                        (result))
-            print("".join(open(self.stderr).readlines()))
             # If the compiler crashed fail the test
-            if 'Compiler Bug' in open(self.stderr).readlines():
+            if 'Compiler Bug' in open(self.outputs["stderr"]).readlines():
                 sys.exit(FAILURE)
-        # check if we expect the p4 compilation of the p4 file to fail
-        expected_error = isError(self.options.p4filename)
+
+        # Check if we expect the p4 compilation of the p4 file to fail
+        expected_error = is_err(self.options.p4filename)
         if expected_error:
-            # we do, so invert the result
+            # We do, so invert the result
             if result == SUCCESS:
                 result = FAILURE
             else:
                 result = SUCCESS
         return result, expected_error
 
     def generate_model_inputs(self, stffile):
-        ''' To override '''
-        ''' Parses the stf file and creates a .pcap file with input packets.
+        """ To override """
+        """ Parses the stf file and creates a .pcap file with input packets.
             It also adds the expected output packets per interface to a global
             dictionary.
             TODO: Create pcap packet data structure and differentiate
-            between input interfaces.'''
+            between input interfaces."""
```

Comment:
In general you don't want TODO in the comment describing the function. That should be a separate comment.